octoprint_mrbeam/iobeam/laserhead_handler.py

```diff
@@ -38,8 +39,8 @@ def __init__(self, plugin):
         self._load_laser_heads_file()  # Loads correction_settings, last_used_lh_serial and lh_cache
 
         self._current_used_lh_serial = self._last_used_lh_serial
-        self._current_used_lh_model = self._last_used_lh_model
-        self._current_used_lh_model_id = None
+        self._current_used_lh_model_string = self._last_used_lh_model_string
+        self._current_used_lh_model_id = self._last_used_lh_model_id
```

Comment:
That is nice, thank you for adding the consistency :)

Reply:
However, if the model id and the model string are refering to the same resource, then the derivative values should be observing the source and shouldn't be variables themselves.

It would be more prudent that `self._current_used_lh_model_id` is the source variable and ` self._current_used_lh_model_string` becomes a function which uses the ` self._current_used_lh_model_id`. You can turn it into a `property` which allows it to be used the same way you would normally do with an object class property.

```python
@property
def  _current_used_lh_model_string(self):
    match self._current_used_lh_model_id:
        case 0: return "D"
        case 1: return "S"
        case _: raise ValueError("Unknown laserhead model ID {!r}".format(self._current_used_lh_model_id))

Reply:
Same goes for `_last_used_lh_model_string` - this way you never forget to update the origin resource and the derivative values :)

Reply:
Actually I would recommend to use a mapping instead of a match / case.